test/Models/TerriaSpec.ts

```diff
@@ -324,157 +303,196 @@ describe("Terria", function() {
-      expect(newTerria.splitPosition).toEqual(0.7);
-      expect(newTerria.workbench.itemIds).toEqual(["itemABC"]);
-
-      const newModel1 = <WebMapServiceCatalogItem>(
-        newTerria.getModelById(BaseModel, "itemABC")
-      );
-      expect(newModel1).toBeDefined();
-      expect(newModel1.splitDirection).toEqual(
-        <any>ImagerySplitDirection.RIGHT
-      );
+      it("opens and loads members of shared open groups", async function() {
+        const group = <WebMapServiceCatalogGroup>(
+          terria.getModelById(BaseModel, "groupABC")
+        );
+        await viewState.viewCatalogMember(group);
+        expect(group.isOpen).toBe(true);
+        expect(group.members.length).toBeGreaterThan(0);
+        const shareLink = buildShareLink(terria, viewState);
+        await newTerria.updateApplicationUrl(shareLink);
+        await newTerria.loadInitSources();
+        const newGroup = <WebMapServiceCatalogGroup>(
+          newTerria.getModelById(BaseModel, "groupABC")
+        );
+        expect(newGroup.isOpen).toBe(true);
+        expect(newGroup.members).toEqual(group.members);
+      });
     });
 
-    it("opens and loads members of shared open groups", async function() {
-      const group = <WebMapServiceCatalogGroup>(
-        terria.getModelById(BaseModel, "groupABC")
-      );
-      await viewState.viewCatalogMember(group);
-      expect(group.isOpen).toBe(true);
-      expect(group.members.length).toBeGreaterThan(0);
-      const shareLink = buildShareLink(terria, viewState);
-      await newTerria.updateApplicationUrl(shareLink);
-      await newTerria.loadInitSources();
-      const newGroup = <WebMapServiceCatalogGroup>(
-        newTerria.getModelById(BaseModel, "groupABC")
-      );
-      expect(newGroup.isOpen).toBe(true);
-      expect(newGroup.members).toEqual(group.members);
+    describe("using story route", function() {
+      beforeEach(async function() {
+        // These specs must run with a Terria constructed with "appBaseHref": "/"
+        terria.updateParameters({
+          storyRouteUrlPrefix: "test/stories/"
+        });
+      });
+      it("sets playStory to 1", async function() {
+        await terria.updateApplicationUrl(
+          new URL("story/my-story", document.baseURI).toString()
+        );
+        expect(terria.userProperties.get("playStory")).toBe("1");
+      });
+      it("correctly adds the story share as a datasource", async function() {
+        await terria.updateApplicationUrl(
+          new URL("story/my-story", document.baseURI).toString()
+        );
+        expect(terria.initSources.length).toBe(1);
+        expect(terria.initSources[0].name).toMatch(/my-story/);
+        if (!isInitData(terria.initSources[0]))
+          throw new Error("Expected initSource to be InitData from my-story");
+
+        expect(toJS(terria.initSources[0].data)).toEqual(
+          (await (await fetch("test/stories/TerriaJS%20App/my-story")).json())
```

Comment:
nitpick: one other way to do this might be to `require()` the json file.

Reply:
Yeah, I was going to do that, but I couldn't work out how to get webpack to import `my-story` (which lacks the `.json` file extension) as a JSON file